src/communication/network_interface_sr.c

```diff
@@ -10813,13 +10819,22 @@ sflashback_get_summary (THREAD_ENTRY * thread_p, unsigned int rid, char *request
   char *ptr;
   char *start_ptr;
 
+  char *num_ptr;		//pointer in which 'number of summary' is located
+  int tmp_num;
+
   LC_FIND_CLASSNAME status;
 
   FLASHBACK_SUMMARY_CONTEXT context = { LSA_INITIALIZER, LSA_INITIALIZER, NULL, 0, 0, };
 
   time_t start_time = 0;
   time_t end_time = 0;
 
+  char *classname = NULL;
+
+  /* *INDENT-OFF* */
+  std::queue <OID> classoid_queue; //queue to return class OID in the order of the class names received from the user
```

Comment:
You don't push to and pop from the queue while running, but just collect and iterate in order. I recommend you to use just `std::vector`. push_back() where `classoid_queue.push (classoid);`, just iterating as you did before when `classoid_queue.front ();`

Reply:
I don't remember the context, so I'm asking. Do you need to maintain two data structures containing the same elements, `classoid_queue` and `context.classoid_set`. Can't we just keep one and make the other from it?

Reply:
I'll use vector to replace queue and set.